perfkitbenchmarker/linux_benchmarks/iperf_benchmark.py

```diff
@@ -98,63 +126,191 @@ def _RunIperf(sending_vm, receiving_vm, receiving_ip_address, thread_count,
+                                         timeout=FLAGS.iperf_runtime_in_seconds +
+                                         timeout_buffer)
+
+    multi_thread = re.search((r'\[SUM\]\s\d+\.?\d+-\d+\.?\d+\ssec\s+\d+\.?\d+\s+MBytes\s+\d+\.?\d+'
+                              r'\s+Mbits\/sec\s+(?P<write>\d+)\/(?P<err>\d+)\s+(?P<pps>\d+)\s+pps'),
+                             stdout)
+
+    match = re.search(r'UDP buffer size: (?P<buffer_size>\d+\.\d+)\s+(?P<buffer_unit>\w+)', stdout)
+    buffer_size = float(match.group('buffer_size'))
+    buffer_size_unit = match.group('buffer_unit')
+    datagram_size = int(re.findall(r'(\d+)\sbyte\sdatagrams', stdout)[0])
+    ipg_target = float(re.findall(r'IPG\starget:\s(\d+.?\d+)', stdout)[0])
+    ipg_target_unit = str(re.findall(r'IPG\starget:\s\d+.?\d+\s(\S+)\s', stdout)[0])
+
+    if multi_thread:
+      # Write, Err, PPS
+      write = int(multi_thread.group('write'))
+      err = int(multi_thread.group('err'))
+      pps = int(multi_thread.group('pps'))
+
+    else:
+      # Write, Err, PPS
+      match = re.search(r'\d+\s+Mbits\/sec\s+(?P<write>\d+)\/(?P<err>\d+)\s+(?P<pps>\d+)\s+pps', stdout)
+      write = int(match.group('write'))
+      err = int(match.group('err'))
+      pps = int(match.group('pps'))
+
+    # Jitter
+    jitter_array = re.findall(r'Mbits\/sec\s+(\d+\.?\d+)\s+[a-zA-Z]+', stdout)
+    jitter_avg = sum(float(x) for x in jitter_array) / len(jitter_array)
+
+    jitter_unit = str(re.findall(r'Mbits\/sec\s+\d+\.?\d+\s+([a-zA-Z]+)', stdout)[0])
+
+    # total and lost datagrams
+    lost_datagrams_array = re.findall(r'Mbits\/sec\s+\d+\.?\d+\s+[a-zA-Z]+\s+(\d+)\/\s+\d+\s+\(', stdout)
+    total_datagrams_array = re.findall(r'Mbits\/sec\s+\d+\.?\d+\s+[a-zA-Z]+\s+\d+\/\s+(\d+)+\s+\(', stdout)
+
+    lost_datagrams_sum = sum(int(x) for x in lost_datagrams_array)
+    total_datagrams_sum = sum(int(x) for x in total_datagrams_array)
+
+    # out of order datagrams
+    out_of_order_sum = 0
+    out_of_order_array = re.findall(r'(\d+)\s+datagrams\sreceived\sout-of-order', stdout)
+    if out_of_order_array:
+      out_of_order_sum = sum(int(x) for x in out_of_order_array)
+
+    thread_values = re.findall(r'\[SUM].*\s+(\d+\.?\d*).Mbits/sec', stdout)
+    if not thread_values:
+      # If there is no sum you have try and figure out an estimate
+      # which happens when threads start at different times.  The code
+      # below will tend to overestimate a bit.
+      thread_values = re.findall(r'\[.*\d+\].*\s+(\d+\.?\d*).Mbits/sec\s+\d+\/\d+', stdout)
+
+      if len(thread_values) != thread_count:
+        raise ValueError(f'Only {len(thread_values)} out of {thread_count} iperf threads reported a'
+                         ' throughput value.')
+
+    total_throughput = 0.0
+    for value in thread_values:
+      total_throughput += float(value)
+
+    metadata = {
+        # The meta data defining the environment
+        'receiving_machine_type': receiving_vm.machine_type,
```

Comment:
Please centralize these rather than copy and paste.

Reply:
centralized the duplicate part of the metadata